File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/application/port/inbound/newsletter/SubscribeToNewsletterUseCase.java
package io.miragon.example.application.port.inbound.newsletter;

import io.miragon.example.domain.Email;
import io.miragon.example.domain.Name;
import io.miragon.example.domain.NewsletterId;
import io.miragon.example.domain.SubscriptionId;

public interface SubscribeToNewsletterUseCase {

    record Command(
        Email email,
        Name name,
        NewsletterId newsletterId
    ) {}

    SubscriptionId subscribe(Command command);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/application/port/inbound/newsletter/SubscribeToNewsletterUseCase.kt
package io.miragon.example.application.port.inbound.newsletter

import io.miragon.example.domain.Email
import io.miragon.example.domain.Name
import io.miragon.example.domain.newsletter.NewsletterId
import io.miragon.example.domain.SubscriptionId

interface SubscribeToNewsletterUseCase {

    fun subscribe(command: Command): SubscriptionId

    data class Command(
        val email: Email,
        val name: Name,
        val newsletterId: NewsletterId
    )
}